src/libraries/Common/src/Extensions/TypeNameHelper/TypeNameHelper.cs

```diff
@@ -110,14 +110,14 @@ private static void ProcessGenericType(StringBuilder builder, Type type, Type[]
             int offset = 0;
             if (type.IsNested)
             {
-                offset = type.DeclaringType.GetGenericArguments().Length;
+                offset = type.DeclaringType!.GetGenericArguments().Length;
```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.type.declaringtype?view=net-5.0

`type.DeclaringType` is not null when Nested or IsArray